tfhe/src/high_level_api/integers/signed/compressed.rs

```diff
@@ -68,9 +72,13 @@ where
     /// Decompress to a [FheInt]
     ///
     /// See [CompressedFheInt] example.
-    pub fn decompress(self) -> FheInt<Id> {
-        let inner = self.ciphertext.into();
-        FheInt::new(inner)
+    pub fn decompress(&self) -> FheInt<Id> {
+        FheInt::new(match &self.ciphertext {
+            CompressedSignedRadixCiphertext::Seeded(ct) => ct.into(),
```

Comment:
ct.decompress() once the refactor is done ?

Reply:
Done